### PR TITLE
Add auto-deploy script

### PR DESCRIPTION
**Pull request by sophalHong**
# Auto Deployment (minio + velero)

This script is created to automatically install velero on Kubernetes.  
First, it deploys minio as deployment and service on Kubernetes.  
Then, it deploys velero server by executing velero CLI (install).  

## Tested environment
ubuntu 18.04.3 LTS  
Kubernetes v1.15.3  
Docker 19.03.5  

### Install:
`./velero.sh i`

### Uninstall:
`./velero.sh u`

### CODE REVIEW

File: examples/auto-deploy/velero.sh [hunk 1]
```diff
@@ -0,0 +1,233 @@
+#!/bin/bash
+
+ACCESS_KEY=${ACCESS_KEY:-"konkhmer"}
+SECRET_KEY=${SECRET_KEY:-"konkhmer855"}
```
Comment (nrb) [inline examples/auto-deploy/velero.sh]: Are these default values your user names? I think something more generic would be appropriate here.

**Comment by nrb** [comment]
Thanks for this PR! While I definitely like the idea of having a quick, self-contained script for getting Velero with Minio installed fast, I admit I'm unsure about this particular implementation.

I need to think on it a little bit more before I provide a more detailed review.

File: examples/auto-deploy/velero.sh (hunk 1 above)
Comment (sophalHong) [reply examples/auto-deploy/velero.sh]: Okay, those values has been changed.

**Comment by carlisia** [comment]
I ran this script on my Mac, kubervetes version 1.15, and got this error on the backup I tried to create after I installed velero:

```
$  ~/w/s/g/v/v/e/i/bash-with-minio> velero backup logs b1
An error occurred: Get http://10.100.125.100:9000/kubevelero/backups/b1/b1-logs.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20200128%2Fminio%2Fs3%2Faws4_request&X-Amz-Date=20200128T014354Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Signature=e65beac1ebf343f031f06235a2596b965952175ff4055b6d647c291bad70eafe: dial tcp 10.100.125.100:9000: i/o timeout
```

**Comment by sophalHong** [comment]
> I ran this script on my Mac, kubernetes version 1.15, and got this error on the backup I tried to create after I installed velero:
> 
> ```
> $  ~/w/s/g/v/v/e/i/bash-with-minio> velero backup logs b1
> An error occurred: Get http://10.100.125.100:9000/kubevelero/backups/b1/b1-logs.gz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20200128%2Fminio%2Fs3%2Faws4_request&X-Amz-Date=20200128T014354Z&X-Amz-Expires=600&X-Amz-SignedHeaders=host&X-Amz-Signature=e65beac1ebf343f031f06235a2596b965952175ff4055b6d647c291bad70eafe: dial tcp 10.100.125.100:9000: i/o timeout
> ```

Thanks for reviewing
This error is not caused by the script. 
velero backup and restore are still working fine. 
It just could not work well when executing 'logs' or 'describe --details'.

[Problem]
It's caused by being NOT able to access to minio object storage URL (ClusterIP) from velero CLI.
Base on my investigation(not yet reviewing the code), 
when we execute `velero backup logs b1`, velero CLI will send API request to minio object storage URL(clusterIP), which is accessible ONLY from inside the Kubernetes cluster itself.
minio service is created by using service type 'ClusterIP'. (reference: [00-minio-deployment](https://github.com/vmware-tanzu/velero/blob/master/examples/minio/00-minio-deployment.yaml))

[Solution]
Creating minio service by using other type beside 'ClusterIP'.
On my development environment without cloud provider, I could not create and test minio service type 'LoadBalancer' and 'ExternalName', so I have added case of minio service type as 'NodePort', which also solved the problem.

**Comment by carlisia** [comment]
Ah, you're right. Now that you mentioned it, I remembered that after installing, we need to change the Minio Service type from ClusterIP to NodePort to make Minio available outside the cluster. I totally forgot.

**Comment by skriss** [comment]
@sophalHong we appreciate the PR and interest in the project! Unfortunately we've decided that we're not going to accept this into the core Velero repo. While we completely see the value of a one-shot install script like this, we feel that it's too opinionated to be something that we want to maintain as part of the project going forward.

Feel free to publish it in a repo of your own if you still want to make it available for others to discover! And if you're still interested in contributing to the project, please reach out and we can help point you in the right direction. Thanks again.

**Pull request closed without merging**